src/App.tsx

```diff
@@ -156,10 +156,35 @@ function APP() {
     );
   }
 
-  //states
-  const [treatment_units, update_treatment_units] = useState<string[]>([]);
-  const [selected_year, update_selected_year] = useState(2019);
-  const [selected_row, update_selected_row] = useState(null);
+  let opts_year = [2019, 2018, 2017, 2016];
+  const get_valid_year = (year_to_validate: number, valid_years: number[]) => {
+    const max_year = Math.max.apply(null, opts_year);
+    const min_year = Math.min.apply(null, opts_year);
+    return year_to_validate > max_year
+      ? max_year
+      : year_to_validate < min_year
+      ? min_year
+      : year_to_validate;
```

Comment:
I like this:

```suggestion
    return Math.max(min_year, Math.min(number, max_year));
```

Maybe define it as a global helper like this:

`const clamp = (input, min, max) => Math.max(min, Math.min(input, max));`

There is a proposal for Math.clamp but its not ready yet.

Reply:
Love it! <3 so much cleaner now. 
ps: eslint wouldn't let me extend Math.prototype so added a helper function.